Makefile

```diff
@@ -174,6 +174,68 @@ cabal-install-test-accept:
 	rm -rf .ghc.environment.*
 	cd cabal-testsuite && `cabal list-bin cabal-tests` --with-cabal=`cabal list-bin cabal` --hide-successes -j3 --accept ${TEST}
 
+# API generation
```

Comment:
For this much code we should start using gigantic headers to facilitate navigation. E.g.
```
##########################################################################
#                                                                        #
#                              API generation                            #
#                                                                        #
##########################################################################
```
or something...

This applies to other sections as well of course, and is long overdue.

Reply:
fair